monkeys/src/main/kotlin/com/wire/kalium/monkeys/conversation/Monkey.kt

```diff
@@ -96,29 +124,67 @@ class Monkey(
         callback(this)
     }
 
-    fun randomPeer() {
-        this.connectedMonkeys.values.randomOrNull() ?: error("Monkey ${this.user.email} not connected to anyone")
+    fun randomPeer(): Monkey {
+        return MonkeyPool.get(this.connectedMonkeys.randomOrNull() ?: error("Monkey ${this.user.email} not connected to anyone"))
     }
 
-    fun connectTo(anotherMonkey: Monkey) {
-        this.connectedMonkeys[anotherMonkey.user.userId] = anotherMonkey
+    fun randomPeers(userCount: UserCount): List<Monkey> {
+        val count = resolveUserCount(userCount, this.connectedMonkeys.count().toUInt())
+        return this.connectedMonkeys.shuffled().map(MonkeyPool::get).take(count.toInt())
     }
 
-    fun disconnectFrom(anotherMonkey: Monkey) {
-        this.connectedMonkeys.remove(anotherMonkey.user.userId)
+    suspend fun sendRequest(anotherMonkey: Monkey) {
+        this.monkeyState.readyThen {
+            connection.sendConnectionRequest(anotherMonkey.user.userId)
+        }
     }
 
-    suspend fun createConversation(name: String, monkeyList: List<Monkey>, protocol: ConversationOptions.Protocol): MonkeyConversation {
+    suspend fun acceptRequest(anotherMonkey: Monkey) {
+        val self = this
+        this.monkeyState.readyThen {
+            connection.acceptConnectionRequest(anotherMonkey.user.userId)
```

Comment:
This doesn't handle failures. _e.g._ some connection issue or whatever. It would be logged, but would lead to some invalid state where accepting the connection request fails, but we still add it to the local connection list. A growing snowball of invalid states.

Maybe using Kalium as the single source of truth would be best, instead of keeping a copy of the state within the monkey?